java/Nikiforov-Oleksandr/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -0,0 +1,47 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class BinarySearchTree {
+    static class Node {
+        int key;
+        Node left, right;
+
+        public Node(int key) {
+            this.key = key;
+            left = right = null;
+        }
+    }
+
+    private Node root;
+
+    public BinarySearchTree(int value) {
+        root = new Node(value);
+    }
+
+    public void insert(int key) {
+        root = insertRec(root, key);
+    }
+
+    public Node insertRec(Node root, int key) {
+        if (root == null) {
+            root = new Node(key);
```

Comment:
the value can be returned directly, you don't need to assign it to the `root` variable